### ISSUE TITLE
Accuracy/Precision of the future datetime check

### ISSUE THREAD

**Issue by siwhitehouse** [IATI/validator-services#407]
**Brief Description**
The Validator is displaying a future datetime error on an organisation file that has the current date, with the time close to/ahead of the current time. This involves a calculation across timezones

**Severity**
Medium 

**Issue Location**
[IATI Validator - IOM page](https://validator.iatistandard.org/organisation/iom) [now resolved]

**Actual Results/Behaviour**
The Validator displayed an error for the International Organization for Migration's organization file.

![IoMValidatorError](https://github.com/IATI/validator-services/assets/1718090/2492c9b1-1a7e-44c9-b7bb-8bfad1a27cc9)

In the detailed view the error is described as "The last updated datetime of the organisation must not be in the future"

![IoMValidatorErrorDetail](https://github.com/IATI/validator-services/assets/1718090/bc13172f-7817-4928-9a80-8f07092528ef)

My understanding of this error is that the Validator identified the updated file in the Registry at 2023-08-31 10:00 (GMT-5) or 2023-08-31 15:00 (UTC).  The organisation file contains the first line <iati-organisations generated-datetime="2023-08-31T16:10:18.970000+00:00" version="2.03"> which is 31 Aug 2023, 16:10:18. 

So, at the point that it was published the updated datetime was an hour and ten minutes in the future. This could be just ten minutes if I have mishandled the whole GMT/UTC conversion, in which case the error might conceivably be caused by the precision of the Validator.

**Expected Results/Behaviour**
I'm not sure whether this strictly meets a test for having a datetime in the future. In particular:

- should we include the time in the calculation, or is the date sufficient?
- should we allow for any tolerance if an organisation is declaring a last updated datetime that is only a small number of hours in the future?

Also, I'd appreciate a check on my calculations wrt the datetimes in this example please.

**Comment by simon-20** [comment]
Hi @siwhitehouse,

I think your calculation on the timezone is correct.

The file was identified in registry at: 23:00 GMT+8, which is 2023-08-31T15:00:00Z, in UTC time, as you say.

Since the date in the file `last-updated-datetime` attribute was 2023-08-31T16:10:18Z, it was in the future. 

The attribute in question, the `last-updated-datetime` attribute of the `iati-organisation` element, is defined in the schema to be of type `xs:dateTime`, and not just `xs:date`. (https://iatistandard.org/en/iati-standard/203/organisation-standard/iati-organisations/iati-organisation/). So I think the validator was doing the right thing here. 

The questions you raise are sensible to ask, but I think concern how things are formulated in the standard and the ruleset.

As far as this warning is concerned, it would seem that the problem must have arisen due to the last updated date in the file being incorrect, i.e., being set to something in the future. This could have happened for any number of reasons (e.g. misconfigured clock on the computer which produced the file).

An XML comment in the file `org-IOM.xml` says the XML was auto-generated by the IATI CoVE tool. However, this probably just means CoVE was used to convert the data from a spreadsheet. So it may have been that the problem was with the original spreadsheet. Interestingly, the CoVE tool checks the rulesets when it is given an activity file, but it does not seem to check rulesets for organisation files, so errors such as this aren't flagged to the person using CoVE.

If you're in agreement, I think this issue could be closed. The questions you raise probably need discussing, but I think they are ultimately not about the validator tool, but the standard and the rulesets itself.

**Comment by siwhitehouse** [comment]
Thank you for the explication @simon-20 - this clarifies things.

**Comment by simon-20** [comment]
Hi @siwhitehouse,

I've had another look at this, trying to ascertain if this is a widespread problem. I queried all the current validation reports; there are currently no instances of this warning. I also checked to see how many activities had the parallel warning. There are 71 activity files which contain at least one instance of the `last-updated-datetime` is in the future warning. However, they all appear to be general errors: one publisher with 20 activities where the last updated date is set to Jan 1st 2024, and one publisher with 49 files where the last updated date is set to 2032 (presumably a typo), and another one set to 2099--so all genuine errors.

So the Validator appears to be working correctly here.

Simon
